### DISCUSSION TITLE
web3modal and telegram webapp

### DISCUSSION THREAD

**Discussion by kandrukKV** [WalletConnect/walletconnect-monorepo#3986] [Developer Support] [answered]
Hello!

I'm trying to embed web3modal in a Telegram web app.
Now I'm facing a problem.

if I use web3modal V2 then I get an infinite download. The application does not open.
I can click on the link "Try this alternate link" and there is a redirect to "MetaMask", but the window "Connect" does not appear.
If select "Zerion", the connection occurs.

![photo_2023-11-27_12-00-57](https://github.com/WalletConnect/walletconnect-monorepo/assets/54434777/f2104507-3bbf-4878-8ecd-13a5b9244178)

If use web3modal V3, there's no link "Try this alternate link" and I only get an infinite download.

![photo_2023-11-27_17-53-05](https://github.com/WalletConnect/walletconnect-monorepo/assets/54434777/5f298d4e-6b6e-4ae0-9039-4e2bdc2b1454)

What can I do?
Is it possible to include a link "Try this alternate link" in web3modal V3?

**Comment by glitch-txs** [comment, marked as answer]
Hi, I was able to reproduce this issue, I've opened an issue here: https://github.com/WalletConnect/web3modal/issues/1530

Feel free to add anything in the thread. Thanks for reporting! 🤝
